src/main/kotlin/com/manimdsl/frontend/SymbolTable.kt

```diff
@@ -2,30 +2,51 @@ package com.manimdsl.frontend
 
 data class IdentifierData(val type: Type)
 
-interface SymbolTable {
-    fun getTypeOf(identifier: String): Type
-}
+class SymbolTable {
+    private val scopes = mutableListOf<ScopedSymbolTable>(GlobalScopeSymbolTable())
+    private var currentScope: ScopedSymbolTable = scopes[0]
+
+    fun getTypeOf(identifier: String): Type = currentScope.getTypeOf(identifier)
+
+    fun addVariable(identifier: String, type: Type) {
```

Comment:
SymbolTable and ScopedSymbolTable both have addVariable and getTypeOf, can we merge them or at least have an interface (to reduce duplication)?